python/lib/sift_py/report_templates/service.py

```diff
@@ -0,0 +1,139 @@
+
+        if id:
+            report_template = self._get_report_template_by_id(id)
+        elif client_key:
+            report_template = self._get_report_template_by_client_key(client_key)
+
+        if not report_template:
+            raise Exception(f"Report template with client key {client_key} or id {id} not found.")
+
+        return ReportTemplateConfig(
+            name=report_template.name,
+            template_client_key=report_template.client_key,
+            organization_id=report_template.organization_id,
+            tags=[tag.tag_name for tag in report_template.tags],
+            description=report_template.description,
+            rule_client_keys=[rule.client_key for rule in report_template.rules],
+            archived_date=to_timestamp_nanos(report_template.archived_date).to_pydatetime(),
+        )
+
+    def load_report_templates_from_yaml(self, paths: List[Path]) -> List[ReportTemplateConfig]:
+        report_templates = load_report_templates(paths)
+        [
+            self.create_or_update_report_template(report_template)
+            for report_template in report_templates
+        ]
+        return report_templates
+
+    def _get_report_template_by_id(self, report_template_id: str) -> Optional[ReportTemplate]:
+        req = GetReportTemplateRequest(report_template_id=report_template_id)
+        try:
+            res = cast(
+                GetReportTemplateResponse, self._report_template_service_stub.GetReportTemplate(req)
+            )
+            return cast(ReportTemplate, res.report_template) or None
+        except:
+            return None
+
+    def _get_report_template_by_client_key(self, client_key: str) -> Optional[ReportTemplate]:
+        req = GetReportTemplateRequest(client_key=client_key)
+        try:
+            res = cast(
+                GetReportTemplateResponse, self._report_template_service_stub.GetReportTemplate(req)
+            )
+            return res.report_template or None
+        except:
+            return None
+
+    def _create_report_template(self, config: ReportTemplateConfig):
+        client_keys_req = CreateReportTemplateRequestClientKeys(
+            rule_client_keys=config.rule_client_keys
+        )
+        req = CreateReportTemplateRequest(
+            name=config.name,
+            client_key=config.template_client_key,
+            description=config.description,
+            tag_names=config.tags,
+            organization_id=config.organization_id,
+            rule_client_keys=client_keys_req,
+        )
+        self._report_template_service_stub.CreateReportTemplate(req)
+
+    def _update_report_template(
+        self, config: ReportTemplateConfig, report_template: ReportTemplate
+    ):
+        tags = []
+        if config.tags:
+            tags = [ReportTemplateTag(tag_name=tag) for tag in config.tags]
+
+        archived_date = None
+        if config.archived_date:
+            archived_date = to_timestamp_pb(config.archived_date)
```

Comment:
@solidiquis I noticed [this code comment ](https://github.com/sift-stack/sift/blob/main/python/lib/sift_py/_internal/time.py#L34-L35)on this function, do you have a better recommended approach?

Reply:
This should be fine